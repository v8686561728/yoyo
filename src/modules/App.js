import React, { Component } from "react";
import Loadable from "react-loadable";
import { Router, Route, Switch, Redirect } from "react-router-dom";
import Header from "./header/components/Header";
import Footer from "./common/components/Footer";
import Landing from "./landing/Landing";
import history from "./common/components/history";
import LinearProgress from "@material-ui/core/LinearProgress";
import { LocalizeProvider } from "react-localize-redux";
import Login from "../modules/header/components/Login";

export function Loading({ error }) {
  if (error) {
    return (
      <h2
        style={{
          height: "40px",
          background: "#b3d9f7",
          color: "white",
          textAlign: "center",
          verticalAlign: "middle",
          paddingTop: "5px",
          fontSize: "20px",
          fontWeight: "500"
        }}
      >
        Oh nooess! Something went wrong. Try re-loading!
      </h2>
    );
  } else {
    return <LinearProgress color="secondary" />;
  }
}

export const GiftsListContainer = Loadable({
  loader: () => import("./gifts/containers/GiftsListContainer"),
  loading: Loading
});

export const ProfileContainers = Loadable({
  loader: () => import("./user/containers/ProfileContainer"),
  loading: Loading
});

export const GiftShowContainer = Loadable({
  loader: () => import("./gifts/containers/GiftShowContainer"),
  loading: Loading
});

export const GiftsSendContainer = Loadable({
  loader: () => import("./user/containers/GiftsSendContainer"),
  loading: Loading
});

export const GiftsReceivedContainer = Loadable({
  loader: () => import("./user/containers/GiftsReceivedContainer"),
  loading: Loading
});

// const GiftCreateContainer = Loadable({
//   loader: () => import("./gifts/containers/GiftCreateContainer"),
//   loading: Loading
// });

export const AddUpdateForm = Loadable({
  loader: () => import("./admin/components/AddUpdateForm"),
  loading: Loading
});

export const ErrorPage = Loadable({
  loader: () => import("./common/components/ErrorPage"),
  loading: Loading
});

class App extends Component {
  render() {
    return (
      <LocalizeProvider>
        <Header />
        <Router history={history}>
          <Switch>
            {/* <Route path="/giftCards/new" exact component={GiftCreateContainer} /> */}
            <Route exact path="/giftCards/:id" component={GiftShowContainer} />
            <Route exact path="/giftCards" component={GiftsListContainer} />
            <Route exact path="/Profile" component={ProfileContainers} />
            <Route exact path="/GiftsSend" component={GiftsSendContainer} />
            <Route exact path="/Login" component={Login} />
            <Route
              exact
              path="/GiftsReceived"
              component={GiftsReceivedContainer}
            />
            <Route exact path="/AddUpdateForm" component={AddUpdateForm} />
            <Route exact path="/AddUpdateForm/:id" component={AddUpdateForm} />
            <Route exact path="/" component={Landing} />
            <Route exact path="/404" component={ErrorPage} />
            <Redirect to="/404" />
          </Switch>
        </Router>
        <Footer />
      </LocalizeProvider>
    );
  }
}

export default App;
