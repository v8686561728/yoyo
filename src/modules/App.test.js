import React from "react";
import { shallow } from "../enzyme";
import { shallowToJson } from "enzyme-to-json";
import App from "./App";
import { Loading ,GiftsListContainer,ProfileContainers,GiftShowContainer,GiftsSendContainer,GiftsReceivedContainer,AddUpdateForm,ErrorPage} from "./App";


it('should render correctly ErrorPage', () => {
  const tree = shallow(<ErrorPage />);

  expect(tree).toMatchSnapshot();
});

it('should render correctly AddUpdateForm', () => {
  const tree = shallow(<AddUpdateForm />);

  expect(tree).toMatchSnapshot();
});



it('should render correctly GiftsReceivedContainer', () => {
  const tree = shallow(<GiftsReceivedContainer />);

  expect(tree).toMatchSnapshot();
});


it('should render correctly GiftsSendContainer', () => {
  const tree = shallow(<GiftsSendContainer />);

  expect(tree).toMatchSnapshot();
});

it('should render correctly GiftShowContainer', () => {
  const tree = shallow(<GiftShowContainer />);

  expect(tree).toMatchSnapshot();
});

it('should render correctly ProfileContainers', () => {
  const tree = shallow(<ProfileContainers />);

  expect(tree).toMatchSnapshot();
});

it("should render App", () => {
  let wrapper = shallow(<App />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

let loadingProp = {
  error: {}
};
let loadingProp2 = {};
it("should render Loading", () => {
  let wrapper = shallow(<Loading {...loadingProp} />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
it("should render Loading", () => {
  let wrapper = shallow(<Loading {...loadingProp2} />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
it('should render correctly GiftsListContainer', () => {
  const tree = shallow(<GiftsListContainer />);

  expect(tree).toMatchSnapshot();
});