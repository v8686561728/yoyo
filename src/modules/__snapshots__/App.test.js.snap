// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App 1`] = `
<LocalizeProvider>
  <Connect(WithStyles(Header)) />
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/giftCards/:id"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/giftCards"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/Profile"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/GiftsSend"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/Login"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/GiftsReceived"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/AddUpdateForm"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/AddUpdateForm/:id"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/404"
      />
      <Redirect
        push={false}
        to="/404"
      />
    </Switch>
  </Router>
  <WithStyles(Footer) />
</LocalizeProvider>
`;

exports[`should render Loading 1`] = `
<h2
  style={
    Object {
      "background": "#b3d9f7",
      "color": "white",
      "fontSize": "20px",
      "fontWeight": "500",
      "height": "40px",
      "paddingTop": "5px",
      "textAlign": "center",
      "verticalAlign": "middle",
    }
  }
>
  Oh nooess! Something went wrong. Try re-loading!
</h2>
`;

exports[`should render Loading 2`] = `
<WithStyles(ForwardRef(LinearProgress))
  color="secondary"
/>
`;

exports[`should render correctly AddUpdateForm 1`] = `ShallowWrapper {}`;

exports[`should render correctly ErrorPage 1`] = `ShallowWrapper {}`;

exports[`should render correctly GiftShowContainer 1`] = `ShallowWrapper {}`;

exports[`should render correctly GiftsListContainer 1`] = `ShallowWrapper {}`;

exports[`should render correctly GiftsReceivedContainer 1`] = `ShallowWrapper {}`;

exports[`should render correctly GiftsSendContainer 1`] = `ShallowWrapper {}`;

exports[`should render correctly ProfileContainers 1`] = `ShallowWrapper {}`;
